.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,79 @@
+name: Run AD benchmark
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches:
+      - "*"
+
+jobs:
+
+  Build-ad:
+    strategy:
+      matrix:
+        java: [14]
+
+    name: Run Anomaly detection model performance benchmark
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup Java ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      # anomaly-detection
+      - name: Checkout AD
+        uses: actions/checkout@v2
+
+      # dependencies: OpenSearch
+      - name: Checkout OpenSearch
+        uses: actions/checkout@v2
+        with:
+          repository: 'opensearch-project/OpenSearch'
+          path: OpenSearch
+          ref: '1.0.0'
+      - name: Build OpenSearch
+        working-directory: ./OpenSearch
+        run: ./gradlew publishToMavenLocal -Dbuild.snapshot=false
+
+      # dependencies: common-utils
+      - name: Checkout common-utils
+        uses: actions/checkout@v2
+        with:
+          ref: '1.0'
+          repository: 'opensearch-project/common-utils'
+          path: common-utils
+      - name: Build common-utils
+        working-directory: ./common-utils
+        run: ./gradlew publishToMavenLocal -Dopensearch.version=1.0.0
+
+      # dependencies: job-scheduler
+      - name: Checkout job-scheduler
+        uses: actions/checkout@v2
+        with:
+          ref: '1.0'
+          repository: 'opensearch-project/job-scheduler'
+          path: job-scheduler
+
+      - name: Build job-scheduler
+        working-directory: ./job-scheduler
+        run: ./gradlew publishToMavenLocal -Dopensearch.version=1.0.0 -Dbuild.snapshot=false
+      - name: Assemble job-scheduler
+        working-directory: ./job-scheduler
+        run: |
+          ./gradlew assemble -Dopensearch.version=1.0.0 -Dbuild.snapshot=false
+          echo "Creating ../src/test/resources/job-scheduler ..."
+          mkdir -p ../src/test/resources/job-scheduler
+          pwd
+          echo "Copying ./build/distributions/*.zip to ../src/test/resources/job-scheduler ..."
+          ls ./build/distributions/
+          cp ./build/distributions/*.zip ../src/test/resources/job-scheduler
+          echo "Copied ./build/distributions/*.zip to ../src/test/resources/job-scheduler ..."
+          ls ../src/test/resources/job-scheduler
+
+      - name: Build and Run Tests
+        run: |
+          ./gradlew ':test' --tests "org.opensearch.ad.ml.HCADModelPerfTests" -Dtests.seed=2AEBDBBAE75AC5E0 -Dtests.security.manager=false -Dtests.locale=es-CU -Dtests.timezone=Chile/EasterIsland -Dtest.logs=true -Dhcad-benchmark=true
+          ./gradlew integTest --tests "org.opensearch.ad.e2e.DetectionResultEvalutationIT.testDataset" -Dtests.seed=60CDDB34427ACD0C -Dtests.security.manager=false -Dtests.locale=kab-DZ -Dtests.timezone=Asia/Hebron -Dtest.logs=true
```

Comment:
wont these tests just run twice now, as we run integTest in CI.yml workflow? Also are we trying to keep the same seed across all versions?

Reply:
HCADModelPerfTests is only ran in standalone mode. I excluded them in gradle build run. DetectionResultEvalutationIT.testDataset is ran twice. I thought about remove DetectionResultEvalutationIT.testDataset from gradle build, but I feel it is a good signal to check if single stream real time detectors work from end to end. So I kept them. Yes, the seed across versions are the same.

Reply:
Will it be possible to inject seed without changing the API? We don't have access too it then.

Reply:
right now it is not easy to do it without changing the API.